Mathlib/AlgebraicTopology/SimplicialSet/Monoidal.lean

```diff
@@ -0,0 +1,92 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou, Jack McKoen
+-/
+import Mathlib.AlgebraicTopology.SimplicialSet
+import Mathlib.CategoryTheory.ChosenFiniteProducts
+import Mathlib.CategoryTheory.Limits.Shapes.FunctorToTypes
+
+/-!
+# The monoidal category structure on simplicial sets
+
+This file defines an instance of chosen finite products
+for the category `SSet` using the explicit terminal object
+and binary products from the file
+`Mathlib.CategoryTheory.Limits.Shapes.FunctorToTypes`.
+As a result, these constructions give the unit object
+and the tensor product for the monoidal category structure
+on `SSet`.
+
+-/
+
+universe u
+
+open Simplicial CategoryTheory MonoidalCategory
+
+namespace SSet
+
+instance : ChosenFiniteProducts SSet.{u} where
+  terminal := FunctorToTypes.functorEmptyLimitCone _
+  product := FunctorToTypes.binaryProductLimitCone
```

Comment:
In principle it should now be possible to provide such an instance for the category of functors from `C` to `D` whenever `D` has chosen finite products.

Reply:
Thanks for the suggestions. The simp lemmas were slightly more difficult to obtain in general, but it works.